plugins/backstage-plugin-flux/src/hooks/useGetDeployments.ts

```diff
@@ -0,0 +1,47 @@
+import { useApi } from '@backstage/core-plugin-api';
+import { KubernetesApi, kubernetesApiRef } from '@backstage/plugin-kubernetes';
+import { FluxController } from '../objects';
+import { useQuery } from 'react-query';
+import _ from 'lodash';
+
+export const DEPLOYMENTS_PATH =
+  '/apis/apps/v1/namespaces/flux-system/deployments?labelSelector=app.kubernetes.io%2Fpart-of%3Dflux&limit=500';
```

Comment:
What if Flux isn't installed into `flux-system`?

The namespace is hard-coded into the URL here?

I suspect we need to first find the "flux" namespace, with the label `app.kubernetes.io/part-of: flux` ?

Then query that for the deployments?

Alternatively, we could perhaps allow configuration for this?

Reply:
Hey @bigkevmcd! You're right, it's not ready yet :D Started from the premise we're working with flux-system only but next step was to remove the hardcoding. On that topic actually I wanted to check with you and Simon. I am a bit confused b/c, in order to get the list of deployments per cluster, I need the path which has got the namespace in it.
```   
  const deploymentsListsProxyData = await Promise.all(
    clusters?.map(cluster =>
      kubernetesApi.proxy({
        clusterName: cluster.name,
        path: DEPLOYMENTS_PATH,
      }),
    ),
  );
```
The label is part of the deployment but I need to know the namespace before I make the query to get that deployment 🤔

Reply:
Well, when you do `kubectl get -A` it makes a query without including the namespace.

Which would query something like this `/apis/apps/v1/deployments?labelSelector=app.kubernetes.io%2Fpart-of%3Dflux&limit=500`.

Reply:
I think `kubectl get -A` is missing something, I am getting an error back 🤔 . If I don't pass a namespace to `kubectl get deploy -n flux-system -v10 --selector [app.kubernetes.io/part-of=flux](http://app.kubernetes.io/part-of=flux)` I see it defaults to `apis/apps/v1/namespaces/default/deployments?labelSelector=app.kubernetes.io%2Fpart-of%3Dflux&limit=500`

Reply:
Try `kubectl get -A deploy -v10 --selector app.kubernetes.io/part-of=flux`